brick/dockerlib.py

```diff
@@ -180,5 +184,25 @@ def get_image_names_with_dependency_hash(dependency_hash) -> List[str]:
     return [s for s in images.split("\n") if s]
 
 
+def get_image_id_from_dockerfile_contents(dockerfile_contents: str) -> str:
+    from_image_name = [
+        l.split(" ")[1] for l in dockerfile_contents.split("\n") if l.startswith("FROM")
+    ]
+    if len(from_image_name) != 1:
+        raise Exception(f"Did not found a FROM statement in {dockerfile_contents}")
```

Comment:
I think it's worth noting that this line basically prevents using [Docker Multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/).

I'm not sure if that's the use case that we want to support, but if the answer is no, I think we should still reflect that better in the error statement, e.g. have something like this:

``` python
if len(from_image_name) < 1:
        raise Exception(f"Did not found a FROM statement in {dockerfile_contents}")
if len(from_image_name) > 1:
        raise Exception(f"Docker Multi-stage builds are not supported by brick yet: {dockerfile_contents}")
```

If we do want to support it here, we should probably concatenate all `from_image_name` strings and use that instead. And only check for `if len(from_image_name) == 0` before throwing this error of course.

WDYT?

Reply:
Good that you are bringing this up!

Note that Docker is an internal tool used by Brick, so the docker files are actually generated by Brick and the process is opaque to the user.

And as we are not using multi-stage builds, the error handling here is just to be defensive and get an error if we change something inside Brick.

So, I don't think we need to change anything. :)

Reply:
Alright, thanks for the context, feel free to merge the PR then :slightly_smiling_face: